eth2near/eth2near-block-relay-rs/src/relay_errors.rs

```diff
@@ -0,0 +1,71 @@
+use std::error::Error;
+use std::fmt;
+use std::fmt::Display;
+
+#[derive(Debug)]
+pub struct ErrorOnUnwrapSignatureBit();
+
+impl Display for ErrorOnUnwrapSignatureBit {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Error on getting signature bits from sync aggregate!")
+    }
+}
+
+impl Error for ErrorOnUnwrapSignatureBit {}
+
+#[derive(Debug)]
+pub struct SignatureSlotNotFoundError();
+
+impl Display for SignatureSlotNotFoundError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Signature slot not found!")
+    }
+}
+
+impl Error for SignatureSlotNotFoundError {}
+
+#[derive(Debug)]
+pub struct MissSyncAggregationError();
+
+impl Display for MissSyncAggregationError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Sync Aggregation not found. \
+        Beacon Block Body in this blockchain variant doesn't contain sync aggregation. \
+        Please use Altair or The Merge variants"
+        )
+    }
+}
+
+impl Error for MissSyncAggregationError {}
+
+#[derive(Debug)]
+pub struct ExecutionPayloadError();
+
+impl Display for ExecutionPayloadError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Execution Payload not found. \
+        Beacon Block Body in this blockchain variant doesn't contain execution payload. \
+        Please use The Merge variants"
+        )
+    }
+}
+
+impl Error for ExecutionPayloadError {}
+
+#[derive(Debug)]
+pub struct MissSyncCommitteeUpdate();
+
+impl Display for MissSyncCommitteeUpdate {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Sync Committee Update is missing for this Light Client Update"
+        )
+    }
+}
+
+impl Error for MissSyncCommitteeUpdate {}
```

Comment:
I think all of that errors could be grouped in one enum
here is an example:
https://github.com/aurora-is-near/sputnikvm/blob/master/core/src/error.rs#L105
https://github.com/aurora-is-near/aurora-engine/blob/master/engine/src/engine.rs#L117

Reply:
Usually, errors are grouped when there is a function that can return different types of Errors, otherwise having separate types should be good.